Client/Ecosia/Extensions/HomepageViewController+Ecosia.swift

```diff
@@ -7,38 +7,24 @@ import Core
 
 protocol HomepageViewControllerDelegate: AnyObject {
     func homeDidTapSearchButton(_ home: HomepageViewController)
-    func homeDidPressPersonalCounter(_ home: HomepageViewController, completion: (() -> Void)?)
+    func showSettingsWithDeeplink(to destination: AppSettingsDeeplinkOption)
 }
 
-extension HomepageViewController {
-    func configureEcosiaSetup() {
-        personalCounter.subscribe(self) { [weak self] _ in
-            guard let self = self else { return }
-            self.updateTreesCell()
-        }
-
-        referrals.subscribe(self) { [weak self] _ in
-            guard let self = self else { return }
-            self.updateTreesCell()
-        }
-    }
-
-    func updateTreesCell() {
-        guard let impactCell = viewModel.impactViewModel.cell else { return }
-        impactCell.display(treesCellModel, animated: false)
-    }
+protocol SharedHomepageCellDelegate: AnyObject {
+    func openLink(url: URL)
+    func invalidateLayout(at indexPaths: [IndexPath])
+}
 
-    var treesCellModel: NTPImpactCell.Model {
-        .init(impact: User.shared.impact, searches: personalCounter.state!, trees: TreeCounter.shared.treesAt(.init()))
+extension HomepageViewController: SharedHomepageCellDelegate {
+    func openLink(url: URL) {
+        homePanelDelegate?.homePanel(didSelectURL: url, visitType: .link, isGoogleTopSite: false)
     }
-
-    @objc func allNews() {
-        let news = NewsController(items: viewModel.newsViewModel.items, delegate: self)
-        let nav = EcosiaNavigation(rootViewController: news)
-        present(nav, animated: true)
-        Analytics.shared.navigation(.open, label: .news)
+    
+    func invalidateLayout(at indexPaths: [IndexPath]) {
+        let context = UICollectionViewLayoutInvalidationContext()
+        context.invalidateItems(at: indexPaths)
```

Comment:
It isn't since at the moment this is only used for the about section.

I was able to find out that the root cause is actually on `NTPLayout.layoutAttributesForElements`, but have more detailed infos on the issue on a [comment below](https://github.com/ecosia/ios-browser/pull/547#issuecomment-1729664189).